grafana/resource_machine_learning_outlier_detector.go

```diff
@@ -53,9 +52,10 @@ Visit https://grafana.com/docs/grafana-cloud/machine-learning/outlier-detection/
 				Optional:    true,
 			},
 			"datasource_id": {
-				Description: "The id of the datasource to query.",
-				Type:        schema.TypeInt,
-				Optional:    true,
+				Description:  "The id of the datasource to query.",
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ExactlyOneOf: []string{"datasource_uid"},
```

Comment:
It should mention both attributes, I think

```
ExactlyOneOf: []string{"datasource_uid", "datasource_id"},
```

Keeping the test for it should confirm that

Reply:
It threw me too, but apparently not necessary. [Docs say](https://github.com/hashicorp/terraform-plugin-sdk/blob/3495894b26ae61cf11da2669170ff9d1c655896b/helper/schema/schema.go#L271):

> ExactlyOneOf is a set of attribute paths, **including this attribute**, where only one attribute out of all specified can be configured.

I restored the old test which passes it.